internal/host/hostcommands/tang_connectivity_check_cmd.go

```diff
@@ -42,16 +56,60 @@ func (c *tangConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.Ho
 		swag.StringValue(cluster.DiskEncryption.EnableOn) == models.DiskEncryptionEnableOnNone ||
 		swag.StringValue(cluster.DiskEncryption.Mode) == models.DiskEncryptionModeTpmv2 ||
 		!hostutil.IsDiskEncryptionEnabledForRole(*cluster.DiskEncryption, common.GetEffectiveRole(host)) {
-		c.log.Debugf("skipping tangConnectivityCheck for host %s, DiskEncryption config do not require validation here", host.ID.String())
-		return []*models.Step{}, nil
+		c.log.Debugf("skipping tangConnectivityCheck for host %s, cluster DiskEncryption config do not require validation here",
+			host.ID.String())
+		return false
 	}
+	return true
+}
 
+func (c *tangConnectivityCheckCmd) getTangServersFromCluster(cluster common.Cluster, host *models.Host) ([]byte, error) {
+	var (
+		tangServers []byte
+		err         error
+	)
 	request := models.TangConnectivityRequest{TangServers: &cluster.DiskEncryption.TangServers}
-	tangServers, err := json.Marshal(request)
+	tangServers, err = json.Marshal(request)
 	if err != nil {
 		c.log.WithError(err).Errorf("failed to Marshal TangConnectivityRequest with for cluster %s", host.ClusterID)
 		return nil, err
 	}
+	return tangServers, nil
+}
+
+func (c *tangConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models.Step, error) {
+	var (
+		cluster     common.Cluster
+		tangServers []byte
+		err         error
+	)
+
+	if err = c.db.First(&cluster, "id = ?", host.ClusterID).Error; err != nil {
+		c.log.WithError(err).Errorf("failed to fetch cluster %s", host.ClusterID)
+		return nil, err
+	}
+	if swag.BoolValue(cluster.Imported) { // Day 2 imported cluster
```

Comment:
This will not handle a case where a user imported a cluster and updated the cluster resource via API with disk encryption settings. 
@filanov, @avishayt, @omertuc: should we block this via cluster update flow?

Reply:
why would it be blocked? if the user have this information can he update the cluster?

Reply:
Would it not be wise to fall back to getting tang settings from cluster in the cases where it could not be parsed from the ignition, or vice versa?

Some kind of handling for the edge case scenarios, or would this not work?

Reply:
We don't have access to the cluster when it is imported

Reply:
>why would it be blocked? if the user have this information can he update the cluster?

to keep things simple

Reply:
Do we already have the code for blocking the update?  I don't see it here